Syndie flame thrower solution1

Wad67 opened the pull request:
Solution 1 for #33160

Not sure how I'm going to add the various reagents to the flamethrower, but the spray.dm code is not that great for handling flamethrowers.

I also intend to work on another solution, altering the backpack tank version to use the flamethrower procs.

Wad67 committed:
Merge pull request #1 from tgstation/master

Updating fork

Wad67 committed:
Removed syndicate flamethrower backpack, replaced with an upgraded flamethrower with an extra tank.

Removed operator backpack tank.
Added flamethrower/op - Increased release_amount to 0.075 and air_transfer to 7.5
Altered firestarter dufflebag  to include the upgraded flamethrower and a spare plasma tank.

praisenarsie commented on code/game/objects/items/flamethrower.dm:
```diff
@@ -248,3 +248,25 @@
+
+
+
+
```
what's with the whitespace

Supermichael777 commented:
have you forgotten the 50/50 mix of cf3 and flurosulf + salve that monstered the meta a while back?

ShizCalev commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -536,11 +536,12 @@
 
 /obj/item/storage/backpack/duffelbag/syndie/firestarter
-	desc = "A large duffel bag containing New Russian pyro backpack sprayer, a pistol, a pipebomb, fireproof hardsuit, ammo, and other equipment."
+	desc = "A large duffel bag containing an upgraded flamethrower,spare plasma tank, a pistol, a pipebomb, fireproof hardsuit, ammo, and other equipment."
```
with a spare plasma tank*

kevinz000 commented on code/game/objects/items/flamethrower.dm:
```diff
@@ -248,3 +248,25 @@
+
+
+
+/obj/item/flamethrower/op/default_ignite(turf/target, release_amount = 0.075)
```
Or you know use variables instead of hard coding an override to the proc. .

Wad67 replied:
So you would rather it if I had a var for 'release_amount', then apply that to the proc. Then determine the air_transfer.gases[/datum/gas/plasma][MOLES] from said 'release_amount' var?

This would tie in well with what I had planned actually. I initially intended to have the release quantity adjustable with the syndicate flamethrower having higher max values.

kevinz000 commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -536,11 +536,12 @@
 
 /obj/item/storage/backpack/duffelbag/syndie/firestarter/PopulateContents()
 	new /obj/item/clothing/under/syndicate/soviet(src)
-	new /obj/item/watertank/op(src)
```
Please specify you're removing the spray. This does matter .

Wad67 replied:
I did mention this in the commit, but my mistake. I'll endeavor to be more verbose in my pr Description.

Wad67 commented:
I'll try and get this cleaned up at some point today, I'm not overly concerned if this does not get merged. Just attempting to provide a solution for the flamethrower. 

@BeeSting12 Mentioned that the tank/backpack flamethrower was working previously. So it might end up being more worthwhile looking into that rather than attempting to replace it entirely.

Wad67 committed:
Cleaned up trailing whitespace and firestarter desc.

Also switched to using vars for the amount of gas being expelled and updated the proc accordingly.

BeeSting12 commented:
>testing
Well, you're already several steps ahead of some of our coders. *cough* kev *cough*

AnturK commented on code/game/objects/items/flamethrower.dm:
```diff
@@ -197,12 +198,12 @@
 	var/datum/gas_mixture/air_transfer = ptank.air_contents.remove_ratio(release_amount)
 	if(air_transfer.gases[/datum/gas/plasma])
-		air_transfer.gases[/datum/gas/plasma][MOLES] *= 5
+		air_transfer.gases[/datum/gas/plasma][MOLES] *= (gas_release_amount * 100)
```
This seems bit weird to tie directly since you might want better throughput but not the magic plasma multiplication, could you separate the two values please ?

Wad67 replied:
I'd be more than happy to do so.  Although I'm confused as to why.
The multiplication simply serves to release the proportional moles of gas, relative to what was originally in place.

From my understanding, if we increased the release_amount, the air_transfer.gases should also be increased proportionally?

Say if I separated the values, I would have another value called plasma_transfer_quantity, of which I would assign gas_release_amount and then multiply by 100 to achieve the correct value.

If that's all you want me to do, for the purpose of making the code look a bit nicer that's fine.
But I'm not sure what you mean when you say 'magic plasma multiplication'.

AnturK replied:
This code increases the plasma content of the gas sample by five. There's no reason for it except making the burn better, it's not actually relative to the content of the tank hence the "magic". 

Let's say you have tank with 100 moles of plasma in it, if you pour it out all using flamethrower you'll end up with 500 moles of plasma.

AnturK replied:
This could be trivially changed by just removing the increase from the tank of course, but that makes the flamethrower run out very fast.

Wad67 replied:
Alright, I see how this is a problem. What I don't see however is the best solution.
air_transfer is already proportional to the amount of remove_ratio.
And in order to get the flamethrower to be effective, the original code multiplied the amount of plasma produced by 5.

This complicates the problem a bit, ideally the syndicate flamethrower should be better than the standard one. Unfortunately it is already limited by the tank size. 
I have not looked at anything regarding gases or tanks before, so I'm not sure what the best solution would be here.

I'd really rather not multiply the amount of plasma by 5, or whatever number really. Needs to be the exact same amount released, for sanity purposes.
Ideally, the syndicate flamethrower should have a larger tank containing the reagents of the operator backpack flamethrower, but I'm not overly sure how to achieve that. 

At this point I'm thinking of redirecting my efforts towards the backpack flamethrower. The tank has to be larger in order to be more effective, and thus must be worn on the back. Makes sense.

AnturK replied:
Anything happening with this ?

Wad67 replied:
Nah, didn't want to mess around making bigger tanks, or try to figure out what's wrong with the backpack flamethrower.

AnturK reviewed (commented):
Just some minor nitpick.

Cyberboss commented:
Stale, ping or remake
